docs/Projects/TAD/ADRs/0004-software-stack.md

```diff
@@ -0,0 +1,81 @@
+# ADR-0004 Software Stack for TAD
+
+## Context
+
+For building our own TAD solution, we need to choose a software stack. During our earlier
+POCs and market research, we gathered insight and information on technologies to use
+and which not to use.
+
+During further discussions and brainstorm sessions, a software stack was chosen
+that accommodates our needs best.
+
+While more fine grained [requirements](../Existing-Tools/Comparison/requirements) are listed elsewhere,
+some key requirements are:
+
+* the tool is runnable on local dev machines
+* the tool is runnable as user as a local Docker solution
+* the tool is runnable on a cloud platform based on Kubernetes
+* the tool supports multiple organizations, teams and projects
+* the tool supports oath / keycloak
+
+## Assumptions
+
+We stick to one programming language to avoid unneeded complexity. As most AI related
+tooling is written in Python, this is the logical choice for our development as well.
+
+There is no need for a complex or separate web GUI framework. Also to avoid more components and complexity,
+it is preferred to bundle backend and frontend in one solution.
+
+As a government organization, we need to adhere to the digital inclusion standards, like
+[Drempelvrij](https://www.drempelvrij.nl/).
+
+We do not forsee high load or many static files, therefor a separate webserver like
+nginx, apache is not required. Static files will be served directly by the backend.
+
+## Decision
+
+### Programming language
+
+We will use [Python](https://www.python.org/) v3.11.X as programming language
+and [Poetry](https://python-poetry.org/) for dependency management.
+
+### Backend
+
+The Python backend will use the following key dependencies:
+
+* [Granian](https://github.com/emmett-framework/granian) as HTTP server
+* [Pydantic](https://docs.pydantic.dev/latest/) for data validation
+* TODO: decide on [FastAPI](https://fastapi.tiangolo.com/) or
+  [Flask](https://flask.palletsprojects.com/en/3.0.x/)
```

Comment:
or Django?

Reply:
It is open to debate. Today I did a little 'hello to Django' workshop, and it's an extremely powerful and it is packed with features; which is also its drawback, it may be overfeatured for what we need and it feels more steps are required to achieve functionality, which is not a bad thing, but noteworthy.

Reply:
I have the same feeling. 

I'd argue we should go for FastAPI since it is more modern and async (fast).